sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -110,25 +110,30 @@ public SentryTracer(
     this.transactionFinishedCallback = transactionFinishedCallback;
 
     if (idleTimeout != null) {
-      scheduleFinish(idleTimeout);
+      timer = new Timer(true);
+      scheduleFinish();
+    } else {
+      timer = null;
```

Comment:
I guess we can move it from the else-branch to the field initializer, since the field is now not `final` anymore?

Reply:
updated